### PR TITLE
Added roster page

### PR DESCRIPTION
**Pull request by Benson272**
### Administrative Info
Monday Board ID: 1181559666
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added a page for the roster

### Testing
How did you confirm your changes worked? 
- Opened page to see if example users are present
- Added an example alum and switched to the alum tab

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Example users are present
image

Alumni table empty when there are no alumni in database
image

After creating Alumni user table is populated with the alumni
image

### CODE REVIEW

File: frontend/src/pages/Roster.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          .filter((user) => user.role.name !== "Alum")
+          .map((user) => ({
+            ...user,
+            role: user.role.name,
+          }));
+        const alums = data.users
+          .filter((user) => user.role.name === "Alum")
+          .map((user) => ({
+            ...user,
+            role: "Alum",
```
Comment (justinyaodu) [inline frontend/src/pages/Roster.js:62]:
```suggestion
            role: user.role.name,
```
Improves consistency with the line above, and makes the code a bit more flexible if we want to refactor this functionality into a helper function or rename the Alum role.

File: frontend/src/pages/Roster.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          }));
+        setState((prev_state) => ({
+          ...prev_state,
+          loading: false,
+          developers,
+          alumni: alums,
```
Comment (justinyaodu) [inline frontend/src/pages/Roster.js:68]: Is there a reason the two variables have different names (`alumni` vs. `alums`)?

File: frontend/src/pages/Roster.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+}));
+
+const styleButton = {
+  color: "inherit",
+  fontWeight: "900",
+  textDecoration: "underline",
+  textTransform: "none",
+  background: "transparent",
+};
+const deafultButton = {
```
Comment (justinyaodu) [inline frontend/src/pages/Roster.js:30]:
```suggestion
const defaultButton = {
```
Just a spelling issue.

File: frontend/src/pages/Roster.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+      <PageContainer>
+        <LoadingContainer loading={state.loading}>
+          <Button
+            size="large"
+            style={alumni ? deafultButton : styleButton}
+            onClick={() => {
+              setAlumni(false);
+            }}
+          >
+            Active Members
```
Comment (justinyaodu) [inline frontend/src/pages/Roster.js:97]: Can we change this text (and the "Alumni" text) to match the font style on the Figma?

File: frontend/src/pages/Roster.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+                      textAlign: "left",
+                      paddingLeft: "50px",
+                    },
+                    filterCellStyle: {
+                      textAlign: "left",
+                      paddingLeft: "50px",
+                    },
+                  },
+                  {
+                    title: "Role",
```
Comment (justinyaodu) [inline frontend/src/pages/Roster.js:142]: Can we make this display "Previous Role" when viewing alumni, like on the Figma?

**Review by justinyaodu — CHANGES_REQUESTED** [review]
I requested some (hopefully minor) changes. This looks really great overall!

File: frontend/src/pages/Roster.js (hunk 2 above)
Comment (Benson272) [reply frontend/src/pages/Roster.js:68]: I tried having different names for them, but i already have a state variable that keeps track of which table we're looking at (alumni or active member) and I called that var "alumni", so linter gave error for shadow variable. I just ended up changing the state name lol

File: frontend/src/pages/Roster.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+const Roster = () => {
+  const classes = useStyles();
+  const [state, setState] = React.useState({
+    // Initial backend data
+    loading: true,
+    developers: [],
+    alumni: [],
+  });
+  const [selectedRow, setSelectedRow] = React.useState(null);
+  const [alumni, setAlumni] = React.useState(false);
```
Comment (hahsu) [inline frontend/src/pages/Roster.js:45]: Why do you have 2 states with the same name? Maybe this one can be isAlumni instead?

Comment (hahsu) [inline frontend/src/pages/Roster.js:45]: Can we add these into the state on lines 38 - 43, or is there a reason you wanted them to be separate?

**Review by hahsu — CHANGES_REQUESTED** [review]

**Review by hahsu — APPROVED** [review]
lgtm, thanks!

**Review by justinyaodu — APPROVED** [review]
LGTM :)

**Pull request merged**